ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -377,7 +377,7 @@ impl Operand {
         match self {
             Self::LocalVariable(_) => {
                 // The `unwrap` can't fail for a `LocalVariable`.
-                self.to_instr(m).type_(m).unwrap()
```

Comment:
This method is called `type_` because it returns a `Type`, so if you want to rename it to `type_def`, then it should return a `TypeDef`.

But then I'm not sure if we want to rename `Type` to `TypeDef`. That doesn't sit well for some reason.

Reply:
This function is now `is_def` and `type_` in one go: in other words, it returns the type of the definition if there is one.

Reply:
Perhaps `def_type` would be clearer?

Reply:
That's confusing because `def_type()` is already used for instructions that define a type. An operand cannot define a type.

Taking a step back, what's up with `type_`? Is it the trailing underscore that bothers you?

Reply:
Notice that (before this commit) `type_` can *only* be used if `is_def` returns true.

Reply:
Yes, I think I'm ok with all of that, just perhaps not the name `type_def()`.

Reply:
So `def_type` is better?

Reply:
I suggest we keep it `type_`, in line with all of the other `type_` functions on similar types that do the same.

Reply:
Changed in af5864a2.